src/main/java/org/mamute/controllers/ListController.java

```diff
@@ -35,27 +35,33 @@
 	@Inject private NewsDAO newses;
 	@Inject private RecentTagsContainer tagsContainer;
 	@Inject private HttpServletResponse response;
+	@Inject private MessageFactory messageFactory;
 	
 	@Get
-	public void home(Integer p) {
+	public void home(Integer p, SanitizedText message) {
 		Integer page = getPage(p);
 		List<Question> visible = questions.allVisible(page);
 		if (visible.isEmpty() && page != 1) {
 			result.notFound();
 			return;
 		}
-		List<String> tabs = Arrays.asList("voted", "answered", "viewed");
+		List<String> tabs = asList("voted", "answered", "viewed");
 		result.include("tabs", tabs);
 
 		result.include("questions", visible);
 		result.include("totalPages", questions.numberOfPages());
 		result.include("currentPage", page);
 		result.include("currentUser", loggedUser);
+
+		if (message != null && !message.getText().isEmpty()) {
+			result.include("mamuteMessages",
+					asList(messageFactory.build("confirmation", message.getText())));
+		}
```

Comment:
Actually it wouldn't work because I've implemented the delete request with using ajax (so there's no redirection going on). I'll refactor the js to submit a form instead of using ajax and your idea should work.